### ISSUE TITLE
Onfido - Countries in the list are not visible

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#13262]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to any workspace and attempt the Connect Bank Account step
2. Complete All Steps Until Onfido flow triggers in the personal information page 
3. Click on the issuing country dropdown

## Expected Result:
Country Names Should Be Visible

## Actual Result:
Country Names Are Not Visible

## Workaround:
unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.2.35-0
**Reproducible in staging?:** y
**Reproducible in production?:** Can't check
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/205200240-8da01228-09f5-4b92-8c89-b53c67e42908.mov
<img width="960" alt="2" src="https://user-images.githubusercontent.com/43996225/205200261-b80721df-aa3c-42e3-ab79-efbaddf0811f.png">

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @syedsaroshfarrukhdot 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1669903794406069

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by syedsaroshfarrukhdot** [comment]
**Proposal**

We can simply override .onfido-sdk-ui-CountrySelector-CountryDropdown-custom__menu--overlay with desired color as we are currently overriding Onfido styles also to make it look nice 

Also As Discussed In Slack Conversation The Button After Selecting Styles Are Also Missing o We Can Also Fix It in This GH Issue As The Flow is Same 

https://expensify.slack.com/archives/C049HHMV9SM/p1669904053931079

We need to pass  `colorBackgroundDocTypeButton : themeColors.success,` and  change colorBackgroundIcon to `colorBackgroundIcon: themeColors.success,` in BaseOnfidoWeb.js 

After this dropdown and button styling both will be fixed

After Fix :- 

https://user-images.githubusercontent.com/81307212/205205022-bc3a82d6-df3c-4faa-ba4f-5039bf79c5fc.mov

**Comment by MitchExpensify** [comment]
Replicated. Looks unique and noticed due to recent Dark Mode switch:

<img width="375" alt="image" src="https://user-images.githubusercontent.com/36425901/205374317-768e59b2-2340-45ab-b000-244e889cc917.png">

Labeling external

**Comment by Pujan92** [comment]
**Proposal**

1. Change the background color of the custom menu class of onfido
```
.onfido-sdk-ui-CountrySelector-CountryDropdown-custom__menu {
    background-color: #061B09 !important;
}
```

2. Doctype button background change with the icon background
```
colorBackgroundDocTypeButton: themeColors.success,
colorBackgroundIcon: themeColors.transparent,
```

3. Button properties change(remove box shadow on hover,focus) to make it like the expensify button
```
.onfido-sdk-ui-DocumentSelector-DocumentList-option.onfido-sdk-ui-DocumentSelector-DocumentList-optionHoverDesktop:hover {
    box-shadow: none !important;
    background-color: #00C271 !important;
}
.ods-button.-action--primary:focus {
    box-shadow: none !important;
}
```

[a2f2c81c-7e6e-4660-8ab5-d4b11c8ea0e3.webm](https://user-images.githubusercontent.com/14358475/205439241-8747c2eb-0066-4cfd-ad95-8d2bbb537b31.webm)

**Comment by syedsaroshfarrukhdot** [comment]
Update Proposal :- 

Issue with above proposals https://github.com/Expensify/App/issues/13262#issuecomment-1336141513 and My Proposal https://github.com/Expensify/App/issues/13262#issuecomment-1334700583 is as we are hardcoding the color so when when we will implement dark mode switch list will appear like this in the light mode.

![Screen Shot 2022-12-03 at 5 26 17 PM](https://user-images.githubusercontent.com/81307212/205440902-e39ba15a-bb35-411f-a7b0-b3ad78a10495.png)


We need to set appBG color to fix this regression which will be caused in future.

        const root = document.documentElement;
        root.style.setProperty('--listcolor' , themeColors.appBG)

and in index.css  we can use

       .onfido-sdk-ui-CountrySelector-CountryDropdown-custom__menu--overlay{
             background: var(--listcolor) !important;
        }

Also As Discussed In Slack Conversation The Button After Selecting Styles Are Also Missing Styling For DarkMode So We Can Also Fix It in This GH Issue As The Flow is Same

https://expensify.slack.com/archives/C049HHMV9SM/p1669904053931079

We need to pass colorBackgroundDocTypeButton : themeColors.success, and change colorBackgroundIcon to colorBackgroundIcon: themeColors.success, in customUI object in BaseOnfidoWeb.js

if we want to make it exactly like our other buttons we can remove box shadows on the button by setting these classes in index.css also 

      .onfido-sdk-ui-DocumentSelector-DocumentList-option.onfido-sdk-ui-DocumentSelector-DocumentList optionHoverDesktop:hover {
            box-shadow: none !important;
                background-color: #00C271 !important;
        }
       .ods-button.-action--primary:focus {
               box-shadow: none !important;
        }

**Comment by parasharrajat** [comment]
I checked the documentation https://github.com/onfido/onfido-sdk-ui/blob/HEAD/UI_CUSTOMIZATION.md but it does not have any details on customizing the dropdowns. I can not find the source as well. 

With this info, I am fine with @syedsaroshfarrukhdot 's [proposal](https://github.com/Expensify/App/issues/13262#issuecomment-1336152648). 


Things to improve:
1. Use more specific CSS variable names. 


cc: @tylerkaraszewski 

:ribbon: :eyes: :ribbon:  C+ reviewed


--- 

It seems we already have design approval on Slack.

**Comment by syedsaroshfarrukhdot** [comment]
@parasharrajat Thanks, for reviewing sure will keep that in mind and will use more specific CSS variable names while creating PR.

**Comment by MitchExpensify** [comment]
Hired @syedsaroshfarrukhdot on the Upwork job (The correct price will be reflected when payment is made)

Also hired @parasharrajat!

**Comment by syedsaroshfarrukhdot** [comment]
@MitchExpensify Thanks for hiring should I be making a PR ? As I haven't been assigned to job here on GH till now.

**Comment by syedsaroshfarrukhdot** [comment]
Oh my bad. Sure will wait for being assigned to the job by @tylerkaraszewski.

**Comment by MitchExpensify** [comment]
All good!

**Comment by marcaaron** [comment]
>I checked the documentation https://github.com/onfido/onfido-sdk-ui/blob/HEAD/UI_CUSTOMIZATION.md but it does not have any details on customizing the dropdowns. I can not find the source as well.

We can ask them if they can allow it to be customized. The solutions provided are less than ideal imo.

**Comment by marcaaron** [comment]
also I _believe_ we are going to replace the entire Onfido SDK with a custom implementation. 

But not sure when that is happening exactly.

**Comment by syedsaroshfarrukhdot** [comment]
> We can ask them if they can allow it to be customized. The solutions provided are less than ideal imo.

Yes, We can ask them to do so or raise a PR against their repo to support dropdown and button customization ideally it would be the best solution but it would take a lot of  time to do so. Due to this issue Onfido SDK is currently unusable as lists doesn't show in our current dark theme.  In my opinion we can overwrite these styles for now as we are also doing it already. Along with it we can began the process with them to support it. 

https://github.com/Expensify/App/blob/39b2f9924060c7381b2ff37563a5af88360b0d9d/src/components/Onfido/index.css#L1-L13

**Comment by marcaaron** [comment]
>ideally it would be the best solution but it would take a lot of time to do so

Maybe it will. Did we ask to be sure?

>Due to this issue Onfido SDK is currently unusable as lists doesn't show in our current dark theme.

That's a good point. Reverting dark mode doesn't feel like a realistic option.

>In my opinion we can overwrite these styles for now as we are also doing it already.

_We introduced a hack already so we should be allowed to add even more hacks!_

I think this idea exhibits why we should avoid hacks.

**Comment by parasharrajat** [comment]
> We can ask them if they can allow it to be customized. The solutions provided are less than ideal imo.

Yeah, if that can be done. It would be great. 

@syedsaroshfarrukhdot Do you think you can raise an issue on https://github.com/onfido/onfido-sdk-ui for this?

**Comment by syedsaroshfarrukhdot** [comment]
> > We can ask them if they can allow it to be customized. The solutions provided are less than ideal imo.
> 
> Yeah, if that can be done. It would be great. 
> 
> @syedsaroshfarrukhdot Do you think you can raise an issue on https://github.com/onfido/onfido-sdk-ui for this?

Sure, will do that and update.

**Comment by syedsaroshfarrukhdot** [comment]
Sure, I will raise the issue with them and provide an update accordingly.

**Comment by MitchExpensify** [comment]
I'm unsure of the status of this issue, are we moving forward with any proposal here, waiting on a proposal we agree on, or will it be solved when we replace the entire Onfido SDK with a custom implementation?

**Comment by MitchExpensify** [comment]
I'm heading ooo so need to assign a new CM - Reapplying the External label for auto assignment.

**Comment by marcaaron** [comment]
Thanks @syedsaroshfarrukhdot I sent a message to Onfido to try and expedite a response.

**Comment by tylerkaraszewski** [comment]
If anyone wants to take this, they can, but it looks like it's blocked on Onfido currently?

**Comment by syedsaroshfarrukhdot** [comment]
@tylerkaraszewski Yes, we are waiting for a response from Onfido to support it natively. Otherwise with css which would be the ultimate wolution if we don't get a response my solution tests out well and have been reviewed by C+ also.

**Comment by syedsaroshfarrukhdot** [comment]
Just for the reference I found an other issue on there upstream repo where team recommended to overwrite there [CSS](https://github.com/onfido/onfido-sdk-ui/issues/1571#issuecomment-1110964528) until customization are released in the next update.

**Comment by puneetlath** [comment]
Interesting find @syedsaroshfarrukhdot. However, that comment seems to indicate that they added support in version 8.2.0 and I see that they are [now on](https://www.npmjs.com/package//onfido-sdk-ui) version 10.1.0. So could we solve this by simply updating to the latest version?

**Comment by syedsaroshfarrukhdot** [comment]
> Interesting find @syedsaroshfarrukhdot. However, that comment seems to indicate that they added support in version 8.2.0 and I see that they are [now on](https://www.npmjs.com/package//onfido-sdk-ui) version 10.1.0. So could we solve this by simply updating to the latest version?

@puneetlath  No, I don't think we can solve it by updating to latest version. As they currently https://github.com/onfido/onfido-sdk-ui/blob/HEAD/UI_CUSTOMIZATION.md doesn't support customization for the dropdown according to there docs for the latest `version` and the [Comment](https://github.com/onfido/onfido-sdk-ui/issues/2135#issuecomment-1356477746) by team on the issue I created they recognize it as a `bug` and will be discussing it with there design to resolve these issues.

I just pointed out that as `onfido team` also suggested to override there [CSS styles](https://github.com/onfido/onfido-sdk-ui/issues/1571#issuecomment-1110964528) while having such issue in the past. We could do similarly to escalate this process for now as in [My Proposal](https://github.com/Expensify/App/issues/13262#issuecomment-1336152648) as we are currently in #BugZero I guess. 

I will keep following the [Issue](https://github.com/onfido/onfido-sdk-ui/issues/2135) once they release the updated version with dropdown and doctype button text colors we can use those props provided by them and remove CSS customizations.

**Comment by marcaaron** [comment]
Nice, that's great to hear they were responsive!

**Comment by tylerkaraszewski** [comment]
Still waiting on onfido it seems.

**Comment by JmillsExpensify** [comment]
This issue landed on my radar given that it's close to hitting 30 days. Given that a fork isn't in play, and related, we will eventually deprecate the Onfido SDK in the coming months (best ETA we can give, though nothing is certain), then I think we should update the title and HOLD this issue on Onfido.

**Comment by syedsaroshfarrukhdot** [comment]
We have any update from [`Onfido Team`](https://github.com/onfido/onfido-sdk-ui/issues/2135#issuecomment-1368819284) they will adding support for the both options (country selection drop-down background color and document type button background color on hover and active states). They will allow these customizations in next planned release. I have asked them about ETA of next release soon we will have any update the release date.

**Comment by marcaaron** [comment]
Oh Nice! 🎉

**Comment by syedsaroshfarrukhdot** [comment]
We have an update on ETA of the new `Onfido Release` from the `Onfido Team` with support for the both options (country selection drop-down background color and document type button background color on hover and active states). It should be released by 16th Jan.

**Comment by syedsaroshfarrukhdot** [comment]
Sure will do that 🙂

**Comment by syedsaroshfarrukhdot** [comment]
Thanks for reviewing. Will definitely thoroughly test and will make sure that it doesn't break anything after upgrade. `Noted` will not update buttons border colors in this PR.

**Comment by syedsaroshfarrukhdot** [comment]
Will I was working on the PR I noticed a problem with `onfido-sdk-ui` on `android/Chrome`. It gives error on the `android/Chrome` while even `chrome` is updated to latest version. Which makes testing it on `android/Chrome` not possible.

This issue exists on  both`onfido-sdk-ui": "^8.1.0`  and the latest `onfido-sdk-ui": "^10.3.0`

So I believe we should create another GH Issue and open a new upstream issue with onfido to get it resolved.

![WhatsApp Image 2023-01-14 at 6 33 44 PM](https://user-images.githubusercontent.com/81307212/212474405-94c68c1c-8321-4e43-8b28-9b21d8acca5c.jpeg)

Should I move forward with the PR ignoring mWeb/Chrome  ?

CC : @parasharrajat @tylerkaraszewski

**Comment by syedsaroshfarrukhdot** [comment]
> Ok so we can't upgrade it until you have a fix for it.
> 
> Please hold hiring until we have a proposal with proper upgrade instructions.

@parasharrajat It existed on `onfido-sdk-ui": "^8.1.0` I have tested it. After upgrading to `onfido-sdk-ui": "^10.3.0` this issue still presist. So It should not be considered as bug related to this PR and should be treated separately.

**Comment by syedsaroshfarrukhdot** [comment]
@parasharrajat I apologize for any confusion caused by my previous message. Upon further investigation, it appears that the issue was not with the Onfido SDK, but rather with my test setup. I had been running the Onfido SDK on a physical device using my IP address and port number, which caused the error "Unsupported Browser" to appear.

Upon further testing, I discovered that the Onfido SDK was functioning properly on both the staging and production environments.

To resolve the issue, I utilized an emulator and ran the command "adb reverse tcp:8080 tcp:8080" in the command prompt. This allowed me to access the Onfido SDK through localhost:8080, instead of through my IP address and port number. This resolved the issue and the Onfido SDK is now functioning properly.

It is unclear why this issue occurred when using my IP address and port number, but utilizing an emulator and accessing the Onfido SDK through localhost resolved the issue.

mWeb/Android :-

https://user-images.githubusercontent.com/81307212/212488004-8a356e19-5db4-4d6a-91e0-0412f71e535f.mov

Based on my testing, it appears that all platforms are functioning properly and the pull request is ready for submission. Just need to get assigned to the job and I will raise PR in next few hours.

**Comment by syedsaroshfarrukhdot** [comment]
@parasharrajat have you tested it can we move forward with my proposal?

**Comment by puneetlath** [comment]
@tylerkaraszewski that good with you?

**Comment by tylerkaraszewski** [comment]
That's fine with me.

**Comment by mallenexpensify** [comment]
Can someone either room `hold` from the issue title or the `Help Wanted` label? I'm unsure which is best

**Comment by syedsaroshfarrukhdot** [comment]
Thanks for assigning @puneetlath. 

PR is [UP](https://github.com/Expensify/App/pull/14353) for review

**Comment by syedsaroshfarrukhdot** [comment]
Can we get the PR reviewed by another engineer if @tylerkaraszewski is busy or OOO. As today is 3rd business day after it will not be eligible for timeliness bonus. As it is awaiting CME engineer review since yesterday to get merged.It have been reviewed by C+.

**Comment by tylerkaraszewski** [comment]
Reviewing today.

**Comment by mallenexpensify** [comment]
Popping in quick cuz I also think @puneetlath who's BZ is OOO til Monday.  Since there was a delay in the internal review, we'll take that into consideration when addressing a potential timeliness bonus/penalty.  Please continue working on with urgency to get it finished @syedsaroshfarrukhdot

**Comment by syedsaroshfarrukhdot** [comment]
I think GH automation failed on this issue. PR was deployed to production 3 days ago. Can we add awaiting payment label here. As it should be ready to payout on 31-01-2023.

**Comment by mallenexpensify** [comment]
Hit production on 1/24 - https://github.com/Expensify/App/pull/14414#issuecomment-1402991988
Updated title to pay on 1/31
Removed `Reviewing` label so `Overdue` will work
Reason automation might not have worked is because, in the OP of the PR, there wasn't a space after `$`

**Comment by syedsaroshfarrukhdot** [comment]
@mallenexpensify bumping for payment as it is ready for payment. It was merged in 3 business days. So it is also eligible for 50% bonus.

**Comment by mallenexpensify** [comment]
@syedsaroshfarrukhdot I'm showing more than 3 biz days, am I missing something? 
[Assigned Jan 16th](https://github.com/Expensify/App/issues/13262#event-8240830544) - Monday
[PR Merged](https://github.com/Expensify/App/pull/14414#event-8327715977) - Jan 21st - Saturday.

**Comment by syedsaroshfarrukhdot** [comment]
@mallenexpensify Yes, You are referring to follow up PR to downgrade onfido-sdk version to 10.3.0 to remove vulnerability which was created by @tylerkaraszewski.

[Assigned Jan 16](https://github.com/Expensify/App/issues/13262#event-8240830544) - Monday
[PR Merged Jan 19](https://github.com/Expensify/App/pull/14353#event-8315131647) - Thursday

**Comment by mallenexpensify** [comment]
Ah ha!  Thanks for clarifying
Both @parasharrajat and @syedsaroshfarrukhdot have been paid $6000 which include the timeliness bonus.  Thanks for sticking with this one and for the prompt PR submission and review.   

For updating the regression test steps, I propose adding a step 3 and 4 to [Workspace - Add bank account - Onfido (Offline)](https://expensify.testrail.io/index.php?/cases/view/1971173&group_by=cases:section_id&group_order=asc&display_deleted_cases=0&group_id=229065)
3. Click on the issuing country dropdown
4 Verify the country names show.  

@parasharrajat can you address the top three check boxes below? 

BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] The PR that introduced the bug has been identified. Link to the PR:
- [ ] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. Link to the GH issue for creating the test here:

**Comment by syedsaroshfarrukhdot** [comment]
> Ah ha!  Thanks for clarifying
> 
> Both @parasharrajat and @syedsaroshfarrukhdot have been paid $6000 which include the timeliness bonus.  Thanks for sticking with this one and for the prompt PR submission and review.   
> 
> 
> 
> For updating the regression test steps, I propose adding a step 3 and 4 to [Workspace - Add bank account - Onfido (Offline)](https://expensify.testrail.io/index.php?/cases/view/1971173&group_by=cases:section_id&group_order=asc&display_deleted_cases=0&group_id=229065)
> 
> 3. Click on the issuing country dropdown
> 
> 4 Verify the country names show.  
> 
> 
> 
> @parasharrajat can you address the top three check boxes below? 
> 
> 
> 
> BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> 
> 
> - [ ] The PR that introduced the bug has been identified. Link to the PR:
> 
> - [ ] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
> 
> - [ ] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
> 
> - [ ] A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. Link to the GH issue for creating the test here:
> 
> 

@mallenexpensify Sorry, but i just checked I haven't received payment. You ended contract for reporting bonus with me only.

**Comment by syedsaroshfarrukhdot** [comment]
@mallenexpensify No problem, Thank you for responding it so promptly 😀

**Comment by puneetlath** [comment]
@mallenexpensify has everyone been paid? 

If so, given that this was an Onfido issue, I think we're good to close this out.

**Comment by mallenexpensify** [comment]
Thanks y'all, closing!

**Comment by marcaaron** [comment]
@parasharrajat @syedsaroshfarrukhdot Do you all think the effort put in here is worth a total payout of $12,000? I don't mean any disrespect, but seems a bit inflated to me. Correct me if I'm wrong. But I think all of the price increases here were directly due to time spent waiting for a response from Onfido.

Before the first increase, we agreed to look at an upstream fix and sent a message to Onfido [here](https://github.com/Expensify/App/issues/13262#issuecomment-1350198531). I guess I'm not sure what our logic is for continuing to bump the price of the issue if we were waiting for a response from a 3rd party. There isn't any solution that could have been proposed during that time that would have made us change course, right?

I think the spirit behind bumping the issue is that it incentivizes people to propose solutions. But this case doesn't really make sense to me. Thanks!

**Comment by syedsaroshfarrukhdot** [comment]
@marcaaron,I was under the impression that this is how things are done at Expensify. I spent a month carefully monitoring the issue and working closely with upstream partners to continually merge updates and resolve the issue quickly. This was driven by the $4,000 payout and the desire to receive a 50% bonus, which I felt would motivate me to act efficiently. However, if this understanding was incorrect, I think it should have been made clear beforehand. My belief was that the high payout was meant to recognize the efforts of contributors who work on multiple issues but proposals get rejected oftenly and there is only one winning proposal at the end to compensate them for all their hard work in resolving Expensify issues in future also with prompt and efficient solutions.

I have noticed instances where contributors receive similar amounts for relatively straightforward fixes, even when the price increases automatically during proposal review or there are delays in the initial review process.

In general, Expensify payouts are higher than typical freelance rates to incentivize quick and effective resolution of issues.

**Comment by marcaaron** [comment]
Just to clarify a few things here:

- It's just my opinion that it looks inflated at this stage.
- Not suggesting that we change anything about the payout. I think what's done is done.
- Not suggesting there was any wrongdoing in this case on either of your parts. 
- Trying to gather retrospective feedback so we can improve our program.

Thanks for the help on this issue and for sharing your thoughts with me!
